dali/pipeline/executor/async_pipelined_executor.h

```diff
@@ -36,9 +36,11 @@ class DLL_PUBLIC AsyncPipelinedExecutor : public PipelinedExecutor {
                                            size_t bytes_per_sample_hint, bool set_affinity = false,
                                            int max_num_stream = -1,
                                            int default_cuda_stream_priority = 0,
-                                           QueueSizes prefetch_queue_depth = QueueSizes{2, 2})
+                                           QueueSizes prefetch_queue_depth = QueueSizes{2, 2},
+                                           bool get_memory_stats = false)
```

Comment:
I am against converging to infinity with the number of arguments in Executor's or Pipeline's constructor.

I see that's already done in the pipeline, do you think that Executor can get the same treatment with a setter for this option?

Reply:
> Also, we need to keep in mind to not do stats gathering in perf-critical places.

I don't think it does now.

Reply:
Done